### PR TITLE
Cache client capabilities upon first request in chart renderer

### PR DESCRIPTION
**Pull request by afritzler**
**What this PR does / why we need it**:
Cache client capabilities upon first request in chart renderer

### CODE REVIEW

File: pkg/chartrenderer/default.go [hunk 1]
```diff
@@ -38,16 +37,22 @@ const notesFileSuffix = "NOTES.txt"
...
+	renderer     *engine.Engine
+	capabilities *chartutil.Capabilities
 }
 
 // New creates a new DefaultChartRenderer object. It requires a Kubernetes client as input which will be
 // injected in the Tiller environment.
 func New(client kubernetes.Client) ChartRenderer {
+	sv, err := client.Clientset().Discovery().ServerVersion()
+	if err != nil {
+		logger.Logger.Errorf("failed to get kubernetes server version %v", err)
```
Comment (rfranzke) [inline pkg/chartrenderer/default.go]: We should return an error here instead of logging and continuing.

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/chartrenderer/default.go (hunk 1 above)
Comment (afritzler) [reply pkg/chartrenderer/default.go]: I will propagate the error. However this will affect a lot of files.

**Comment by afritzler** [comment]
Alright, I updated the PR. The error should now be propagated.

File: pkg/operation/botanist/dns.go [hunk 2]
```diff
@@ -50,12 +50,15 @@ func (b *Botanist) DestroyExternalDomainDNSRecord() error {
 // DeployDNSRecord kicks off a Terraform job of name <alias> which deploys the DNS record for <name> which
 // will point to <target>.
 func (b *Botanist) DeployDNSRecord(terraformerPurpose, name, target string, purposeInternalDomain bool) error {
+	op, err := terraformer.NewFromOperation(b.Operation, terraformerPurpose)
```
Comment (rfranzke) [inline pkg/operation/botanist/dns.go]: Please replace ALL occurrences of `op` with `tf` (`terraformer.New()` and `terraformer.NewFromOperation` return a `*Terraformer` object, and not an `*Operation` object).

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/operation/botanist/dns.go (hunk 2 above)
Comment (afritzler) [reply pkg/operation/botanist/dns.go]: Good point.

Comment (afritzler) [reply pkg/operation/botanist/dns.go]: Done!

**Review by rfranzke — APPROVED** [review]
LGTM

**Pull request merged**